### PR TITLE
Changed instance dtype in toml to float64; in construction, zfill for…

### PR DESCRIPTION
**Pull request by zorge69**
…mtype to four characters

## Pull Request submission

*Insert detailed bullet points about your changes here!*

*Insert any instructions to help the reviewer, e.g. "install new requirements from `environment.yml`"*

*Let the reviewer know what data files are needed (and if applicable, where they are to be found)

#### Closes or fixes

* Detail the ticket(s) you are closing with this PR
Closes #

#### Code

- [ ] **Code runs** The code runs on my machine and/or CDSW
- [ ] **Conflicts resolved** There are no conflicts (I have performed a rebase if necessary)
- [ ] **Requirements** My/our code functions according to the requirements of the ticket
- [ ] **Dependencies** I have updated the environment yaml so it includes any new libraries I have used
- [ ] **Configuration file updated** any high level parameters that the user may interact with have been put into the config file (and imported to the script)
- [ ] **Clean Code**
    - [ ] Code is as [PEP 8]([url](https://peps.python.org/pep-0008/)) compliant as I can humanly make it
    - [ ] Code passess flake8 linting check
    - [ ] Code adheres to [DRY](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)
- [ ] **Type hints** All new functions have [type hints ](https://mypy.readthedocs.io/en/stable/cheat_sheet_py3.html)

#### Documentation

Any new code includes all the following forms of documentation:

- [ ] **Function Documentation** Docstrings within the function(s')/methods have been created
    - [ ] Includes `Args` and `returns` for all major functions
    - [ ] The docstring details data types
- [ ] **Updated Documentation**: User and/or developer working doc has been updated

#### Data
- [ ] All data needed to run this script is available in Dev/Test
- [ ] All data is excluded from this pull request
- [ ] Secrets checker pre-commit passes

#### Testing
- [ ] **Unit tests** Unit tests have been created and are passing _or a new ticket to create tests has been created_

---

# Peer Review Section

- [ ] All requirements install from (updated) `environment.yaml`
- [ ] Documentation has been created and is clear - **check the working document**
- [ ] Doctrings ([Google format](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html)) have been created and accurately describe the function's functionality
- [ ] Unit tests pass, or if not present _a new ticket to create tests has been created_
- [ ] **Code runs** The code runs on reviewer's machine and/or CDSW

#### Final approval (post-review)

The author has responded to my review and made changes to my satisfaction.
- [ ] **I recommend merging this request.**

---

### Review comments

*Insert detailed comments here!*

These might include, but not exclusively:

- bugs that need fixing (does it work as expected? and does it work with other code
  that it is likely to interact with?)
- alternative methods (could it be written more efficiently or with more clarity?)
- documentation improvements (does the documentation reflect how the code actually works?)
- additional tests that should be implemented (do the tests effectively assure that it
  works correctly?)
- code style improvements (could the code be written more clearly?)
- Do the changes represent a change in functionality so the version number should increase? Start a discussion if so.
- As a review you can generates the same outputs from running the code

Your suggestions should be tailored to the code that you are reviewing.
Be critical and clear, but not mean. Ask questions and set actions.

### CODE REVIEW

File: src/construction/construction.py [hunk 1]
```diff
@@ -78,6 +78,12 @@ def run_construction(
     validate_data_with_schema(construction_df, schema_path)
     construction_df = construction_df.dropna(axis="columns", how="all")
 
+    # Convert formtype to "0001" or "0006"
+    construction_df["formtype"] = construction_df["formtype"].astype('str')
+    construction_df["formtype"] = construction_df["formtype"].str.zfill(4)
+    construction_df["formtype"] = construction_df["formtype"].astype('category')
+
```
Comment (JenCheshire) [inline src/construction/construction.py]: On the network, there's something not quite right with how it's dealing with nulls:
image

Comment (JenCheshire) [reply src/construction/construction.py]: This is from the imputation qa file:
image

File: config/construction_schema.toml [hunk 2]
```diff
@@ -430,7 +430,7 @@ Possible_categorical_Values = ["nan"]
 
 [instance]
 Description = "nan"
-Deduced_Data_Type = "Int64"
+Deduced_Data_Type = "float64"
```
Comment (JenCheshire) [inline config/construction_schema.toml:433]: short to long doesn't seem to be carrying through

**Review by JenCheshire — COMMENTED** [review]
The DAP fixes don't seem to work on the network.  I can look at these in more detail if helpful

**Review by JenCheshire — APPROVED** [review]
@zorge69 I have added a function to deal with the formtype - this sorts out the issues, as discussed earlier, so I have approved the PR

**Comment by zorge69** [comment]
I converted dtype of instance to float64 in construction.toml 
This is needed for construction.py on df.update, see [line 132](https://github.com/ONSdigital/research-and-development/blob/1b5d94d70b54d09e9700d70c7997702177fbfc94/src/construction/construction.py#L132)
`    updated_snapshot_df.update(construction_df)
`

**Pull request merged**